cmd/lotus-provider/config.go

```diff
@@ -1,61 +1,180 @@
 package main
 
 import (
+	"context"
+	"errors"
 	"fmt"
+	"os"
+	"regexp"
+	"strings"
 
+	"github.com/BurntSushi/toml"
+	"github.com/kr/pretty"
 	"github.com/urfave/cli/v2"
+
+	"github.com/filecoin-project/lotus/lib/harmony/harmonydb"
+	"github.com/filecoin-project/lotus/node/config"
 )
 
 var configCmd = &cli.Command{
 	Name:  "config",
-	Usage: "Manage node config",
+	Usage: "Manage node config by layers. The layer 'base' will always be applied. ",
 	Subcommands: []*cli.Command{
 		configDefaultCmd,
 		configSetCmd,
 		configGetCmd,
+		configListCmd,
+		configViewCmd,
 	},
 }
 
 var configDefaultCmd = &cli.Command{
 	Name:  "default",
-	Usage: "Print default system config",
+	Usage: "Print default node config",
 	Flags: []cli.Flag{
 		&cli.BoolFlag{
 			Name:  "no-comment",
 			Usage: "don't comment default values",
 		},
 	},
 	Action: func(cctx *cli.Context) error {
-		fmt.Println("[config]\nstatus = Coming Soon")
-		// [overlay.sealer1.tasks]\nsealer_task_enable = true
+		c := config.DefaultLotusProvider()
+
+		cb, err := config.ConfigUpdate(c, nil, config.Commented(!cctx.Bool("no-comment")), config.DefaultKeepUncommented(), config.NoEnv())
+		if err != nil {
+			return err
+		}
+
+		fmt.Print(string(cb))
+
 		return nil
 	},
 }
 
 var configSetCmd = &cli.Command{
-	Name:  "set",
-	Usage: "Set all config",
+	Name:      "set",
+	Usage:     "Set a config layer or the base",
+	ArgsUsage: "a layer's name",
 	Action: func(cctx *cli.Context) error {
-		fmt.Println("Coming soon")
+		args := cctx.Args()
+		if args.Len() != 1 {
+			return errors.New("must have exactly 1 arg for the file name")
+		}
+		db, err := makeDB(cctx)
+		if err != nil {
+			return err
+		}
+
+		fn := args.First()
+		bytes, err := os.ReadFile(fn)
```

Comment:
Can we make the arg optional, and use stdin when no args are set?

Reply:
Then what would we munge into the layer's name? (today myLayer.toml becomes myLayer)

Reply:
Could have a separate `--title` flag (much like kubo has `--stdin-name` for ipfs-add), and make that flag non-optional when pulling input from stdin